tests/test_property.py

```diff
@@ -1734,3 +1734,227 @@ def test_slice_for_box(example_model_with_properties):
+
+    ntg1 = np.ones(shape = (6, 10, 10))
+    ntg2 = np.zeros(shape = (6, 10, 10))
+
+    fine_pc.add_cached_array_to_imported_list(cached_array = ntg1,
+                                              source_info = '',
+                                              keyword = 'ntg',
+                                              discrete = False,
+                                              property_kind = 'net to gross ratio',
+                                              realization = 1)
+
+    fine_pc.add_cached_array_to_imported_list(cached_array = ntg2,
+                                              source_info = '',
+                                              keyword = 'ntg',
+                                              discrete = False,
+                                              property_kind = 'net to gross ratio',
+                                              realization = 2)
+    fine_pc.write_hdf5_for_imported_list()
+    fine_pc.create_xml_for_imported_list_and_add_parts_to_model()
+
+    # Act
+    coarse_pc.extend_imported_list_copying_properties_from_other_grid_collection(other = fine_pc,
+                                                                                 coarsening = fc,
+                                                                                 realization = 1)
+    coarse_pc.write_hdf5_for_imported_list()
+    coarse_pc.create_xml_for_imported_list_and_add_parts_to_model()
+
+    # Assert
+    assert len(coarse_pc.parts()) == numc + 1
+
+    newparts = [part for part in coarse_pc.parts() if coarse_pc.citation_title_for_part(part) == 'ntg']
+    assert len(newparts) == 1
+    result = coarse_pc.cached_part_array_ref(newparts[0])
+
+    assert_array_almost_equal(np.ones(shape = (3, 5, 5)), result)
+
+
+def test_import_ab_properties(example_model_with_properties, test_data_path):
+    # Arrange
+    model = example_model_with_properties
+    pc = rqp.GridPropertyCollection(grid = model.grid())
+    init_num = len(pc.parts())
+    ab_facies = os.path.join(test_data_path, 'facies.ib')
+    ab_ntg = os.path.join(test_data_path, 'ntg_355.db')
+
+    # Act
+    pc.import_ab_property_to_cache(ab_facies, keyword = 'ab_facies', discrete = True, property_kind = 'discrete')
+
+    pc.import_ab_property_to_cache(ab_ntg, keyword = 'ab_ntg', discrete = False, property_kind = 'net to gross ratio')
+    pc.write_hdf5_for_imported_list()
+    pc.create_xml_for_imported_list_and_add_parts_to_model()
+
+    # Assert
+    assert len(pc.parts()) == init_num + 2
+    # Check NTG array
+    ntg = [part for part in pc.parts() if pc.citation_title_for_part(part) == 'ab_ntg'][0]
+    ntg_array = pc.cached_part_array_ref(ntg)
+    assert np.min(ntg_array) > 0.4
+    assert np.max(ntg_array) < 0.7
+    assert np.allclose(np.mean(ntg_array), 0.550265)
+
+    # Check facies array
+    facies = [part for part in pc.parts() if pc.citation_title_for_part(part) == 'ab_facies'][0]
+    facies_array = pc.cached_part_array_ref(facies)
```

Comment:
Could you please add a check that the facies part is flagged as discrete. I think  the following should do it:
assert not pc.continuous_for_part(facies)

Reply:
... and I guess something similar to check that the ntg part is continuous, above

Reply:
Oh and whilst you are at it, why not check that the property kind is as expected too!